### DISCUSSION TITLE
switch_bucket() sets grad_req to 'write' by default and overwrites other settings

### DISCUSSION THREAD

**Discussion by ArneNx** [apache/incubator-mxnet#10904] [Q&A]
https://github.com/apache/incubator-mxnet/blob/251ae71a20d8318ab20f4c19520f74b881fdf3ff/python/mxnet/module/bucketing_module.py#L353

I want to set the ``grad_req`` for all my parameters to ``'add'`` by using specifying it in the ``bind()`` call on my top-level module. 
The problem is that I use a ``BucketingModule``. So when I call ``forward``, it automatically calls the ``switch_bucket`` method, which then calls ``bind`` again without specifying ``grad_req`` and therefore setting it to ``'write'``. 

Is this intended behavior? Can I somehow work around this while still using bucketing?

**Comment by vdantu** [comment]
@sandeep-krishnamurthy : Could you please tag this "Question", "Python"

@ArneNx : You could post this question on "discuss.mxnet.io". You would receive a wider audience.
